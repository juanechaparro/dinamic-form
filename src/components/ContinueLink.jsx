import React from 'react'

export const ContinueLink = () => {
  return (
    <LinK>ContinueLink</Link>
  )
}
