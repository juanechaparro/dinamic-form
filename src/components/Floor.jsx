
export const Floor = () => {
  return (
    <div>Floor</div>
  )
}
