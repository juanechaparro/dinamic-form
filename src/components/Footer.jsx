import logo from "../assets/logo.png";
import "../styles/Footer.css";
export const Footer = () => {
  return (
    <footer className="footer">
      <div className="footer-logo">
        <img src={logo} alt="Logo" />
      </div>
      <div className="footer-text">
        <p>2023 HabiÂ® Todos los derechos reservados.</p>
      </div>
    </footer>
  );
};
