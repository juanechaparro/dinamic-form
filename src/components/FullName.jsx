
export const FullName = () => {
  return (
    <div>FullName</div>
  )
}
