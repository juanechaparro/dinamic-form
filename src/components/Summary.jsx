import { useDispatch, useSelector } from "react-redux";
import PropTypes from "prop-types";
import Swal from "sweetalert2";
import { uiCloseModal } from "../redux/actions/ui";
import { useNavigate } from "react-router-dom";
import { clearForm } from "../redux/actions/form";
import "../styles/Summary.css";

export const Summary = ({ lastStep = false }) => {
  const { form } = useSelector((state) => state);
  const { StepsDataLoading, StepsData } = form;
  const Navigate = useNavigate();
  const dispatch = useDispatch();
  const closeModal = () => {
    dispatch(uiCloseModal());
  };
  const sendForm = () => {
    //validate all steps
    const StepsToComplete = [];
    StepsData.forEach((step) => {
      if (!form[step.component] && step.component !== "SummaryStep") {
        StepsToComplete.push(`Paso ${step.order} ${step.name} `);
      }
    });
    if (StepsToComplete.length > 0) {
      Swal.fire({
        title: "Error!",
        text: `Debe completar los siguientes pasos: ${StepsToComplete.join(
          ", "
        )}`,
        icon: "error",
        confirmButtonText: "Ok",
      });
    } else {
      Swal.fire({
        title: "Formulario enviado!",
        text: "Gracias por completar el formulario",
        icon: "success",
        confirmButtonText: "Ok",
      });
      localStorage.clear();
      dispatch(clearForm());
      Navigate("/");
    }
  };
  if (StepsDataLoading) {
    return <div className="summary-loading">...Loading</div>;
  }
  return (
    <div className={`summary ${lastStep ? "summary_lastStep" : ""}`}>
      <h2 className="summary-title">Resumen de datos :</h2>
      <div className="summary-content">
        {StepsData.map((step) => {
          const formData = form[step.component];
          const hasFormData = formData && formData.value;
          const hasError = formData && formData.error;

          return (
            hasFormData && (
              <div key={step.component} className="summary-step">
                <div className="summary-step-header">
                  <span className="summary-step-number">
                    Paso {step.order} {step.name} :
                  </span>
                  <span className="summary-step-value">{formData.value}</span>
                </div>
                {hasError && (
                  <div className="summary-error">Error: {formData.error}</div>
                )}
                <div className="summary-description">
                  Descripci√≥n: {step.description}
                </div>
              </div>
            )
          );
        })}
      </div>
      {lastStep ? (
        <button className="summary-send-button" onClick={sendForm}>
          Enviar
        </button>
      ) : (
        <button className="summary-close-button" onClick={closeModal}>
          Cerrar
        </button>
      )}
    </div>
  );
};

Summary.propTypes = {
  lastStep: PropTypes.bool,
};
