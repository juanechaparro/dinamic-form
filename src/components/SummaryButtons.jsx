export const SummaryButtons = (lastStep, sendForm, closeModal) => {
  return lastStep ? (
    <button className="summary-send-button" onClick={sendForm}>
      Enviar
    </button>
  ) : (
    <button className="summary-close-button" onClick={closeModal}>
      Cerrar
    </button>
  );
};
