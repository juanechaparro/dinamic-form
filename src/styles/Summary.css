.summary-modal-container{
    width: 300px;
    height: 350px;
}
.summary {
    border: 1px solid #ccc;
    border-radius: 4px;
    padding: 20px;
    background-color: #f9f9f9;
    margin: 0px 0px 20px 30px;
  }
.summary_lastStep{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 90vh;
    margin:25px;
}
.summary-send-button{
    margin-bottom: 30px;
    margin-top: 5px;
    padding: 10px;
    background-color: #8004fc;
    color: #ffffff;
    border: none;
    border-radius: 4px;
    font-size: 16px;
    font-weight: bold;
    cursor: pointer;
    transition: background-color 0.3s ease;
}
  .summary-loading {
    font-size: 18px;
    text-align: center;
    color: #888;
  }
  
  .summary-title {
    font-size: 24px;
    margin-bottom: 20px;
    margin-right: 10px;
  }
  
  .summary-content {
    margin-top: 10px;
  }
  
  .summary-step {
    margin-bottom: 20px;
    padding-bottom: 10px;
    border-bottom: 1px solid #ccc;
  }
  
  .summary-step-header {
    display: flex;
    justify-content: space-between;
    margin-bottom: 10px;
    font-weight: bold;
  }
  
  .summary-step-number {
    font-size: 18px;
    color: #8004fc;
  }
  
  .summary-step-value {
    font-size: 18px;
  }
  
  .summary-error {
    color: red;
    margin-bottom: 5px;
  }
  
  .summary-description {
    font-size: 14px;
    color: #555;
  }
  .summary-close-button {
    display: block;
    width: 100%;
    padding: 10px;
    background-color: #8004fc;
    color: #ffffff;
    border: none;
    border-radius: 4px;
    font-size: 16px;
    font-weight: bold;
    cursor: pointer;
    transition: background-color 0.3s ease;
  }
  
  .summary-close-button:hover {
    background-color: #6c02d1;
  }
  
  @media (min-width: 768px) {
    .summary-close-button {
      display: none;
    }
  }
  
  /* Media queries para dispositivos m√≥viles */

  @media (max-width: 768px) {
    .summary {
      padding: 10px;
    }
  
    .summary-title {
      font-size: 20px;
      margin-bottom: 15px;
    }
  
    .summary-step {
      margin-bottom: 15px;
      padding-bottom: 5px;
    }
  
    .summary-step-header {
      flex-direction: column;
      align-items: flex-start;
      margin-bottom: 5px;
    }
  
    .summary-step-number {
      font-size: 16px;
      margin-bottom: 5px;
    }
  
    .summary-step-value {
      font-size: 16px;
    }
  
    .summary-description {
      font-size: 12px;
    }
  }