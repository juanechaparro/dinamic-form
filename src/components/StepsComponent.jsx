import PropTypes from 'prop-types'
import { Address } from "./Address"
import { Floor } from "./Floor"
import { FullName } from "./FullName"
import { stepComponentsNames } from '../stepComponentsNames'
import { SummaryStep } from './SummaryStep'
import {Options} from './Options'
import { Email } from './Email'

export const StepsComponent = ({step}) => {
        switch (step.component) {
            case stepComponentsNames.FullName:
                return <FullName />
            case stepComponentsNames.Address:
                return <Address />
            case stepComponentsNames.Floor:
                return <Floor />
            case stepComponentsNames.Summary:
                return <SummaryStep />
            case stepComponentsNames.Options:
                return <Options />
            case stepComponentsNames.Email:
                return <Email />
            default:
                return <div>Component not found</div> 
        }
}
StepsComponent.propTypes = {
    step: PropTypes.object.isRequired
}
