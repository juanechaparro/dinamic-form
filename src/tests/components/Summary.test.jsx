import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { store } from "../../redux/store";
import { render } from "@testing-library/react";
import { Summary } from "../../components/Summary";

describe("pruebas en Summary", () => {
  test("RenderizaciÃ³n correcta del componente", () => {
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Summary />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
