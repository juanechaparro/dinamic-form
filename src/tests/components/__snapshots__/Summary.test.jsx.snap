// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en Summary Renderizaci√≥n correcta del componente 1`] = `
<div>
  <div
    class="summary-loading"
  >
    ...Loading
  </div>
</div>
`;
